pkg/istio/rbac/v2/provider.go

```diff
@@ -166,28 +186,57 @@ func (p *v2) ConvertAthenzModelIntoIstioRbac(athenzModel athenz.Model, serviceNa
 				log.Infoln("member expired, skip adding member to authz policy resource, member: ", roleMember.MemberName)
 				continue
 			}
-			namespace, err := common.CheckIfMemberIsAllUsersFromDomain(roleMember, athenzModel.Name)
-			if err != nil {
-				log.Errorln("error checking if role member is all users in an Athenz domain: ", err.Error())
-				continue
-			}
-			if namespace != "" {
-				from_namespace.Source.Namespaces = append(from_namespace.Source.Namespaces, namespace)
-				continue
-			}
-			spiffeName, err := common.MemberToSpiffe(roleMember)
-			if err != nil {
-				log.Errorln("error converting role member to spiffeName: ", err.Error())
-				continue
-			}
-			from_principal.Source.Principals = append(from_principal.Source.Principals, spiffeName)
-			if p.enableOriginJwtSubject {
-				originJwtName, err := common.MemberToOriginJwtSubject(roleMember)
+
+			for _, member := range members {
+				// This is only done for Group Members to check the expiry and system disabled
+				// at a Group Member level after doing a check for the entire role
```

Comment:
Why don't we need to do this for the single role members as well? Little bit confused here as to why we need to differentiate, didn't the previous logic do the check for the role member?

Reply:
So the check outside the loop is for a role member (which is basically a user/service/group). The expiry/system disabled options are either set for this role member or in case of a group it could also be set at the groupmember (all members in the group) level as well. The loop inside detects the scenario where the member is group member and then makes those additional checks. If it was a role member we have already made those checks outside and it would be repeating that. But, in case of groups there are two levels of checks that are required - 1. for the group present as a rolemember in the role and 2. for the particular group in question.